### ISSUE TITLE
Fails to build, missing std for C code

### ISSUE THREAD

**Issue by falkTX** [harryhaaren/openAV-Fabla2#40]
Relevant part of the build log:
```
[100%] Building C object CMakeFiles/fabla2.dir/src/dsp/sequencer.c.o
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/sequencer.c: In function 'sequencer_reset':
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/sequencer.c:113:2: error: 'for' loop initial declarations are only allowed in C99 or C11 mode
  for( int i = 0; i < MAX_STEPS; i++ ) {
  ^
```

there was also a bunch of these:
```
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/fabla2.cxx: In member function 'void Fabla2::Fabla2DSP::midi(int, const uint8_t*, bool)':
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/fabla2.cxx:275:8: warning: enumeration value 'LV2_MIDI_MSG_INVALID' not handled in switch [-Wswitch]
  switch( lv2_midi_message_type( msg ) ) {
        ^
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/fabla2.cxx:275:8: warning: enumeration value 'LV2_MIDI_MSG_NOTE_PRESSURE' not handled in switch [-Wswitch]
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/fabla2.cxx:275:8: warning: enumeration value 'LV2_MIDI_MSG_CHANNEL_PRESSURE' not handled in switch [-Wswitch]
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/fabla2.cxx:275:8: warning: enumeration value 'LV2_MIDI_MSG_BENDER' not handled in switch [-Wswitch]
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/fabla2.cxx:275:8: warning: enumeration value 'LV2_MIDI_MSG_SYSTEM_EXCLUSIVE' not handled in switch [-Wswitch]
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/fabla2.cxx:275:8: warning: enumeration value 'LV2_MIDI_MSG_MTC_QUARTER' not handled in switch [-Wswitch]
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/fabla2.cxx:275:8: warning: enumeration value 'LV2_MIDI_MSG_SONG_POS' not handled in switch [-Wswitch]
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/fabla2.cxx:275:8: warning: enumeration value 'LV2_MIDI_MSG_SONG_SELECT' not handled in switch [-Wswitch]
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/fabla2.cxx:275:8: warning: enumeration value 'LV2_MIDI_MSG_TUNE_REQUEST' not handled in switch [-Wswitch]
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/fabla2.cxx:275:8: warning: enumeration value 'LV2_MIDI_MSG_CLOCK' not handled in switch [-Wswitch]
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/fabla2.cxx:275:8: warning: enumeration value 'LV2_MIDI_MSG_START' not handled in switch [-Wswitch]
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/fabla2.cxx:275:8: warning: enumeration value 'LV2_MIDI_MSG_CONTINUE' not handled in switch [-Wswitch]
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/fabla2.cxx:275:8: warning: enumeration value 'LV2_MIDI_MSG_STOP' not handled in switch [-Wswitch]
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/fabla2.cxx:275:8: warning: enumeration value 'LV2_MIDI_MSG_ACTIVE_SENSE' not handled in switch [-Wswitch]
/tmp/mbs/modduo/cloud-plugins/build/fabla-1f8dd82066d088f4ffd304f9ef862c1764a7dde3/src/dsp/fabla2.cxx:275:8: warning: enumeration value 'LV2_MIDI_MSG_RESET' not handled in switch [-Wswitch]
```
you should add a "default:" case in there.

**Comment by falkTX** [comment]
Simple patch fix:

```
diff --git a/src/dsp/sequencer.c b/src/dsp/sequencer.c
index 3009e3f..5c7b7a0 100644
--- a/src/dsp/sequencer.c
+++ b/src/dsp/sequencer.c
@@ -105,12 +105,14 @@ void sequencer_reset_playhead( Sequencer* s )
 void sequencer_reset( Sequencer* s )
 {
 	assert( s );
+	int i;
+
 	s->num_steps = 32;
 	s->length = 0;
 	s->cur_step = 0;
 	s->counter = 0;
 
-	for( int i = 0; i < MAX_STEPS; i++ ) {
+	for( i = 0; i < MAX_STEPS; i++ ) {
 		s->steps[i] = 0;
 	}
 }

```

**Comment by harryhaaren** [comment]
Both fixed since 2a22b894fb86abcdd5b98148a24cf9d19b8be1c4, thanks for the patch.
